File: programmers/42576.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {

    unordered_map<string, int> table;
    string answer = "";
    
    for (auto part : participant) table[part]++;
    for (auto part : completion) table[part]--;
    for (auto part : participant) {
        if (table[part] == 1) answer = part;
    }

    return answer;
}

File: programmers/42576.py
def solution(participant, completion):
    dict = {}
    answer = ''

    for part in participant:
        if part in dict:
            dict[part] += 1
        else:
            dict[part] = 1
    for comp in completion:
        dict[comp] -= 1
    for key, val in dict.items():
        if (val > 0):
            answer = key

    return answer